src/Api/PubnubApi/EndPoint/Access/GrantTokenOperation.cs

```diff
@@ -389,6 +401,20 @@ internal async Task<PNResult<PNAccessManagerTokenResult>> GrantAccess()
 
             Dictionary<string, int> uuidPatternBitmaskPermCollection = null;
             atleastOnePermission = FillPermissionMappingWithMaskValues(this.pubnubPatterns.Uuids, atleastOnePermission, out uuidPatternBitmaskPermCollection);
+            
+            Dictionary<string, int> userBitmaskPermCollection = null;
+            Dictionary<string, int> userPatternBitmaskPermCollection = null;
+            if (pubnubResources.Uuids.Count == 0 && pubnubPatterns.Uuids.Count == 0)
+            {
+                atleastOnePermission = FillPermissionMappingWithMaskValues(this.pubnubResources.Users, atleastOnePermission, out userBitmaskPermCollection);
+                atleastOnePermission = FillPermissionMappingWithMaskValues(this.pubnubPatterns.Users, atleastOnePermission, out userPatternBitmaskPermCollection);
```

Comment:
We're overwriting the same variable in multiple places - how does this work?

Reply:
We have "out" param in those calls and code is ensuring that we have atleast one permission in either resources or patterns. atleastOnePermission is not really overwritten because we are sending its value to the method as param. When "atleastOnePermission" is false, next one will check for any perm. If true, no new check.